### PR TITLE
More events 2

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,7 +79,7 @@
     "mocha": "2.3.3",
     "newrelic": "1.22.2",
     "npm-run-all": "1.2.12",
-    "polyfill-service": "2.1.0",
+    "polyfill-service": "2.1.0-beta.1",
```
Comment (ScottDowne) [inline package.json]: Oops, need to remove this.

File: scripts/input-dispatcher.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import dispatcher from '../scripts/dispatcher.js';
```
Comment (ScottDowne) [inline scripts/input-dispatcher.js]: I need to remove this too.

File: tests/react/address-input.tests.jsx [hunk 3]
```diff
@@ -4,61 +4,40 @@ var should = require('should');
 var IntlContext = { messages: {'country': 'c', 'address': 'a', city: 'c', postal_code: 'p', 'state_province': 's'} };
```
Comment (ScottDowne) [inline tests/react/address-input.tests.jsx]: These are busted, but actually they "pass" by having 0 of 0 tests not fail...

Comment (ScottDowne) [inline tests/react/address-input.tests.jsx]: Pretty sure these are already pretty busted on master anyway.

**Comment by alicoding** [comment]
You need to rebase.

**Comment by ScottDowne** [comment]
@alicoding ok rebased.

File: components/address-input.jsx [hunk 4]
```diff
@@ -577,19 +579,36 @@ var ProvinceSelect = React.createClass({
...
     return {
-      valid: true
+      valid: true,
+      country: ""
     };
   },
+  componentDidMount: function() {
+    listener.on("fieldUpdated", this.onFieldUpdated);
+    form.registerField({
+      name: this.props.name,
```
Comment (alicoding) [inline components/address-input.jsx]: You've removed `propTypes` for `onChange` above entirely, but you are still using `this.props.name` here... Can you re-add that back for `name`?

File: components/address-input.jsx [hunk 5]
```diff
@@ -601,35 +620,54 @@ var Country = React.createClass({
...
     return (
-      <CountrySelect ref="countrySelect" name={this.props.name} country={this.props.value} onCountryChange={this.onCountryChange} className={className}/>
+      <CountrySelect ref="countrySelect" name={this.props.name} country={this.state.country} onCountryChange={this.onCountryChange} className={className}/>
     );
   }
 });
 
 var Province = React.createClass({
   mixins: [IntlMixin],
-  propTypes: {
```
Comment (alicoding) [inline components/address-input.jsx]: Here too.

File: components/error-message.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import listener from '../scripts/listener.js';
+
+module.exports = React.createClass({
```
Comment (alicoding) [inline components/error-message.jsx]: Can you add `propTypes` here too?

**Comment by alicoding** [comment]
r+ with small changes

**Comment by alicoding** [comment]
I just reviewed a 1000 lines of code O_o

**Pull request merged**